ahcore/writers.py

```diff
@@ -86,6 +89,8 @@ def __init__(
     def init_writer(self, first_coordinates: GenericArray, first_batch: GenericArray, h5file: h5py.File) -> None:
         """Initializes the image_dataset based on the first tile."""
         if self._is_compressed_image:
+            if self._precision is not None:
```

Comment:
Why is this check performed here? Can't it be done in the constructor already?

Reply:
The subsequent lines of code (from line 93) are executed only if line 91 evaluates to True. So, we have to have the if condition of self._is_compressed_image here. 

Let's say we move it to the constructor of the class, it is possible, but we will have to check the if condition again at line 91.  So, we would be repeating `if self._is_compressed_image` two times. That's why, the precision is checked once while initializing the writer.

Reply:
Discussed this offline and what we have now seems okay.